src/createfunctionsjson.ts

```diff
@@ -0,0 +1,302 @@
+import * as fs from 'fs';
+import * as ts from 'typescript';
+
+const inputFile = process.argv[2];
+const sourceCode = fs.readFileSync(inputFile, 'utf-8');
+const sourceFile = ts.createSourceFile(inputFile, sourceCode, ts.ScriptTarget.Latest, true);
+
+/* tslint:disable:no-reserved-keywords */
+const CUSTOM_FUNCTION = 'customfunction'; // case insensitive @CustomFunction tag to identify custom functions in JSDoc
+const HELPURL_PARAM = 'helpurl';
+
+let errorFound = false;
+let errorLogFile = [];
+let skippedFunctions = [];
+
+const TYPE_MAPPINGS = {
+    [ts.SyntaxKind.NumberKeyword]: 'number',
+    [ts.SyntaxKind.StringKeyword]: 'string',
+    [ts.SyntaxKind.BooleanKeyword]: 'boolean',
+};
+
+/**
+ * Takes the sourcefile and attempts to parse the functions information
+ * @param sourceFile source file containing the custom functions
+ */
+function parseTree(sourceFile: ts.SourceFile): ICFVisualFunctionMetadata[] {
+    const metadata: ICFVisualFunctionMetadata[] = [];
+    console.log("start parse");
+    visit(sourceFile);
+    return metadata;
+
+    function visit(node: ts.Node) {
+        if (ts.isFunctionDeclaration(node)) {
+            if (node.parent && node.parent.kind === ts.SyntaxKind.SourceFile) {
+                const func = node as ts.FunctionDeclaration;
+                const isCF =
+                    ts
+                        .getJSDocTags(func)
+                        .filter(
+                        (tag: ts.JSDocTag) =>
```

Comment:
It would help to pull out functions that can stand alone and remove redundant code. I see many opportunities for this. For example: 
* case-insensitive string comparison -- e.g. `areStringsEqualNoCase(first: string, second: string): boolean { return first.localeCompare(second); }`  -- I think localeCompare() is current best practice
* get the string for a JSDocTag, 
etc.